File: 05-05-25/app/src/main/java/com/example/sharedpreferences/MainActivity.java
package com.example.sharedpreferences;

import android.os.Bundle;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        Storage<T> localStorage = new LocalStorage(this);
        try {
            String valorPrevio = localStorage.getValue("Llave", null);
            if (valorPrevio != null){
                Toast.makeText(
                        this,
                        "Valor previo: " + valorPrevio,
                        Toast.LENGTH_LONG
                ).show();
            }
            localStorage.setValue("Facultad", "Lavve");
        } catch (StorageException e) {
            Toast.makeText(
                    this,
                    "Error al guardar el valor",
                    Toast.LENGTH_LONG
            ).show();

        }
    }
}

File: Mycomposable/app/src/main/java/com/example/mycomposable/MainActivity.kt
package com.example.mycomposable

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.tooling.preview.Preview


class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MessageCard("Hello world!")
        }
    }
}

@Composable
fun MessageCard(name: String)
{
    Text(text = "Hello $name!")
}

@Preview
@Composable
fun PreviewMessageCard() {
    MessageCard("Android")
}